test/unit/connection_test.rb

```diff
@@ -26,6 +26,25 @@ def test_connection_endpoint_raises_uri_error
     end
   end
 
+  def test_connection_passes_env_proxy_by_default
+    @connection.logger.expects(:info).twice
+
+    spy = Net::HTTP.new('example.com', 443)
+    Net::HTTP.expects(:new).with('example.com', 443, :ENV, nil).returns(spy)
+    spy.expects(:get).with('/tx.php', {}).returns(@ok)
+    response = @connection.request(:get, nil, {})
+  end
+
+  def test_connection_does_pass_requested_proxy
+    @connection.logger.expects(:info).twice
+    @connection.proxy_address = "proxy.example.com"
+    @connection.proxy_port = 8080
+    spy = Net::HTTP.new('example.com', 443)
+    Net::HTTP.expects(:new).with('example.com', 443, "proxy.example.com", 8080).returns(spy)
+    spy.expects(:get).with('/tx.php', {}).returns(@ok)
+    response = @connection.request(:get, nil, {})
```

Comment:
Nitpick: we can remove the unused `response ` variable. Same for the other test.